feat(zammad): add config to stack.yml

huzaifaedhi22 opened the pull request:
This PR
- adds Zammad image to stack.yml
- adds zammad.localhost to dev certificates

TODO
- A zammad role in postgres
- A schema for zammad ( Draft )

huzaifaedhi22 committed:
feat(zammad): add config to stack.yml

dargmuesli reviewed (commented):
Dargstack expects each service to have a comment explaining what the service does, compare to the existing services. Also, run `dargstack rgen` to generate an updated README for others to see what services are in use.

huzaifaedhi22 committed:
fix: use postgres role

dargmuesli commented on src/development/stack.yml:
```diff
@@ -295,7 +301,7 @@ services:
       POSTGRES_PASSWORD_FILE: /run/secrets/postgres_password
       POSTGRES_USER_FILE: /run/secrets/postgres_user
-    image: postgis/postgis:17-3.5-alpine
+    image: imresamu/postgis:17-3.5.2-alpine3.21
```
To be reverted

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
       - ../../../reccoom/:/srv/app/
+
+  zammad-elasticsearch:
+    image: bitnami/elasticsearch:${ELASTICSEARCH_VERSION:-8.18.0}
```
```suggestion
    image: bitnami/elasticsearch:8.18.0
```

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
+
+  zammad-elasticsearch:
+    image: bitnami/elasticsearch:${ELASTICSEARCH_VERSION:-8.18.0}
+    restart: always
```
`restart` does not exist for `stack` I believe, see `deploy.restart_policy.condition` (https://docs.docker.com/reference/compose-file/deploy/#restart_policy)

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
     volumes:
       - ../../../reccoom/:/srv/app/
+
+  zammad-elasticsearch:
```
Can elasticsearch be used by Zammad only or can we have it as `elasticsearch` only, similar to `postgres`?

Reading https://github.com/zammad/zammad-docker-compose?tab=readme-ov-file#running-without-elasticsearch it seems like elasticsearch could be excluded from the stack for "small teams". It appears some maintenance may be required for upgrades which I'm not sure how complicated they are. Please check that a bit further.

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
+  zammad-init:
+    image: zammad/zammad-docker-compose:latest
+    restart: on-failure
+    depends_on:
```
`depends_on` does not exist for stacks I believe

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
+    restart: always
+
+  zammad-websocket:
+    image: zammad/zammad-docker-compose:latest
```
```suggestion
    image: zammad/zammad-docker-compose:6.5.0-85
```
maybe?

dargmuesli commented on src/development/stack.yml:
```diff
@@ -551,3 +715,5 @@ volumes:
     # The message queue's data.
     {}
+  zammad_elasticsearch_data: {}
+  zammad_storage: {}
```
```suggestion
  zammad_data: {}
```

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
+      - zammad-railsserver
+      - zammad-websocket
+    environment:
+      - NGINX_SERVER_NAME=zammad.${STACK_DOMAIN}
+      - REDIS_URL=redis://zammad-redis:6379
```
When I apply the same environment variables as for `zammad-websocket` the container starts nginx successfully. Alos, port `8080` needs to be used, not `80`.

dargmuesli commented on src/development/stack.yml:
```diff
@@ -344,6 +350,162 @@ services:
+        target: POSTGRESQL_PASS
+    command: ["zammad-scheduler"]
+
+  zammad-redis:
```
Similar to elasticsearch, can this just be called `redis`?

dargmuesli reviewed (commented):
In general, why do the additions look so different compared [to the source definition](https://github.com/zammad/zammad-docker-compose/blob/master/docker-compose.yml)? Especially in that some services have some labels but not all? Isn't it possible to use `x-shared`?

sthelemann committed:
feat(zammad): use x-shared section

An x-shared section was added to the stack.yml file, similar as in the original docker-compose yaml file to reuse environment settings etc. in several Zammad services.

sthelemann commented:
No idea what is still wrong with the  `[CI / dargstack rgen / Build (ubuntu-latest, 22) (pull_request)` test.

dargmuesli commented:
@sthelemann you have to run `dargstack rgen` to regenerate the README so that it accurately represents the stack contents for people interested in finding our configuration.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/maevsi/stack?pullRequest=202) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/maevsi/stack?pullRequest=202) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: huzaifaedhi22<br/>:x: sthelemann<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/maevsi/stack?pullRequest=202) it.</sub>
